File: backend/schemas/diagram.py
from typing import Literal

from pydantic import BaseModel, Field

from schemas.chat import RetrievedSource


class DiagramRequest(BaseModel):
    message: str = Field(min_length=1, max_length=4000)
    diagram_type: Literal["auto", "dependency", "architecture"] = "auto"
    limit: int = Field(default=50, ge=1, le=200)


class DiagramResponse(BaseModel):
    repository_id: int
    question: str
    diagram_type: Literal["dependency", "architecture"]
    title: str
    description: str
    mermaid: str
    model: str = ""
    sources: list[RetrievedSource] = Field(default_factory=list)

File: frontend/src/types/diagram.ts
export type DiagramRequest = {
  message: string
  limit?: number
}

export type DiagramResponse = {
  repository_id: number
  question: string
  title: string
  description: string
  mermaid: string
  model: string
  sources: Array<{
    file_path: string
    language: string
    chunk_kind: string
    symbol_name: string | null
    start_line: number
    end_line: number
    score: number
    excerpt: string
  }>
}
